src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -181,45 +181,179 @@ def __init__(
-        neighbor_check = f"{field_index} < {field_desc.shape[offset_provider.neighbor_axis.value]}"
-    else:
-        assert isinstance(offset_provider, Dimension)
-        raise NotImplementedError(
-            "Neighbor reductions for cartesian grids not implemented in DaCe backend."
+    lift_context, inner_inputs, inner_outputs = transformer.visit(node.args[0], args=lifted_args)
+    assert len(inner_outputs) == 1
+    inner_out_connector = inner_outputs[0].value.data
+
+    iterator_field_nodes = {}
+    iterator_index_mapping = {}
+    lifted_index_connectors = []
+    for (field_connector, inner_index_table), iterator in filter(
+        lambda x: isinstance(x[1], IteratorExpr), inner_inputs
+    ):
+        iterator_field_nodes[field_connector] = iterator.field
+        for dim, connector in inner_index_table.items():
+            if dim == neighbor_dim:
+                lifted_index_connectors.append(connector)
+            iterator_index_mapping[connector] = iterator.indices[dim]
+
+    input_value_nodes = {
+        connector: expr.value
+        for connector, expr in filter(lambda x: isinstance(x[1], ValueExpr), inner_inputs)
+    }
+
+    input_nodes = {**iterator_field_nodes, **input_value_nodes}
+
+    neighbor_tables = filter_neighbor_tables(transformer.offset_provider)
+    connectivity_names = [connectivity_identifier(offset) for offset in neighbor_tables.keys()]
+
+    parent_sdfg = transformer.context.body
+    parent_state = transformer.context.state
+
+    input_mapping = {
+        connector: create_memlet_full(node.data, node.desc(parent_sdfg))
+        for connector, node in input_nodes.items()
+    }
+    connectivity_mapping = {
+        name: create_memlet_full(name, parent_sdfg.arrays[name]) for name in connectivity_names
+    }
+    array_mapping = {**input_mapping, **connectivity_mapping}
+    symbol_mapping = map_nested_sdfg_symbols(parent_sdfg, lift_context.body, array_mapping)
+
+    nested_sdfg_node = parent_state.add_nested_sdfg(
+        lift_context.body,
+        parent_sdfg,
+        inputs={*input_mapping.keys(), *iterator_index_mapping.keys(), *connectivity_names},
+        outputs={inner_out_connector},
+        symbol_mapping=symbol_mapping,
+    )
+
+    for connectivity_connector, memlet in connectivity_mapping.items():
+        parent_state.add_memlet_path(
+            parent_state.add_access(memlet.data, debuginfo=parent_sdfg.debuginfo),
+            map_entry,
+            nested_sdfg_node,
+            dst_conn=connectivity_connector,
+            memlet=memlet,
         )
 
+    for inner_connector, access_node in input_nodes.items():
+        parent_state.add_memlet_path(
+            access_node,
+            map_entry,
+            nested_sdfg_node,
+            dst_conn=inner_connector,
+            memlet=input_mapping[inner_connector],
+        )
+
+    for inner_connector, access_node in iterator_index_mapping.items():
+        if inner_connector in lifted_index_connectors:
```

Comment:
`lifted_index_connectors` is a `list` lokking might be inefficient, could you use a `set` instead?